File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/data_structures/dp_pair.java
// dp_pair.java

public class dp_pair<First, Second>
{
	public dp_pair(First a_first, Second a_second)
	{
		set_first(a_first);
		set_second(a_second);
	}

	public void set_first(First a_first)
	{
		my_first = a_first;
	}

	public First get_first()
	{
		return my_first;
	}

	public void set_second(Second a_second)
	{
		my_second = a_second;
	}

	public Second get_second()
	{
		return my_second;
	}

	private First  my_first;
	private Second my_second;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/builder/solution/DpPair.m
% DpPair.m

classdef DpPair < handle
	properties
		First  % any type
		Second % any type
	end

	methods
		% constructor:
		function theDpPair = DpPair(aFirst, aSecond)
			theDpPair.First  = aFirst;
			theDpPair.Second = aSecond;
		end
	end
end
